### DISCUSSION TITLE
Aggregating CONUS404 hourly data to daily values.

### DISCUSSION THREAD

**Discussion by amsnyder** [hytest-org/hytest#412] [Data Processing and Analysis]
When aggregating the CONUS404 hourly data to daily values, it is important to keep track of whether the variable you are using is an instantaneous value or an aggregated value. Aggregated variables will have an attribute attached which indicates the period over which the data was aggregated, called `integration_length`.

If you were processing the `PREC_ACC_NC` variable, for example, the `intregration_length` attribute indicates that the data value is `accumulated over prior 60 minutes`. Therefore, to get a daily summary, you will want to sum the values for `01:00:00` of that day to `00:00:00` of the next day. You can do this with xarray by running a line of code like `ds_daily_sum = ds.PREC_ACC_NC.resample(time="24H", base=1, loffset='-1H').sum()`. The `base=1` parameter shifts the daily aggregation ahead one hour (by default, it would use `00:00:00` to `23:00:00` of that day), and the `loffset='-1H'` adjusts the time stamp on the resulting aggregated dataset back to `00:00:00`, rather that using the first time stamp it used of `01:00:00`.

If you are aggregating an instantaneous data variable, you do not need to do this offset, and you can calculate your aggregation with something like: `ds_daily_avg = ds.T2.resample(time="24H").mean()`.

**Comment by sfoks** [comment]
Very useful info!! Thank you!!
